### PR TITLE
Migrate contract functionality

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Migrate contract functionality from market-creator to trader repository. Market creator will use the corresponding packages from trader as dependencies.

This is required to avoid a lot of code duplication when trying to implement the redeem functionalityon market creator.

### CODE REVIEW

**Review by 0xArdi — APPROVED** [review]

**Review by Adamantios — CHANGES_REQUESTED** [review]
LGTM, there's an unrelated issue on CI. Please skip and open an issue if you do not address it.

Did you run the agent after the changes?

**Comment by jmoreira-valory** [comment]
> LGTM, there's an unrelated issue on CI. Please skip and open an issue if you do not address it.
> 
> Did you run the agent after the changes?

I didn't run the agent, but given that I am adding contract calls that are not referenced anywhere in this service, I assume there will not be any issue.

File: packages/packages.json [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
+        "connection/valory/ledger/0.19.0": "bafybeia47rr37ianvwsh77tjjpv3nwif5sywhhy2fbdshnz4a2icwln76a",
+        "connection/valory/p2p_libp2p_client/0.1.0": "bafybeihge56dn3xep2dzomu7rtvbgo4uc2qqh7ljl3fubqdi2lq44gs5lq",
+        "connection/valory/ipfs/0.1.0": "bafybeigjtca6b5wwiobl37es3wllzt6ai3qef5gqxzfy5fmmc65jkbnec4",
+        "contract/valory/gnosis_safe_proxy_factory/0.1.0": "bafybeiehjccqvhrcarhahhyyrshaifoipfqwvpxjucpucslp22l2wc3sl4",
+        "contract/valory/gnosis_safe/0.1.0": "bafybeiegkl6zrbvlbpwmoziw4hfkocjfqdqphkmpgow7ovsmv7bwa3f4pe",
+        "contract/valory/service_registry/0.1.0": "bafybeieeuruq7dp4b6souetv5ce7wlm52pw7vyrxuhqa7kuy5m3rwwxks4",
+        "skill/valory/transaction_settlement_abci/0.1.0": "bafybeifpnkwgwpzz6uwrvfgurm26allr6shjfbp7bfbrxwy64sw3nf3fsa",
         "contract/valory/multisend/0.1.0": "bafybeig5byt5urg2d2bsecufxe5ql7f4mezg3mekfleeh32nmuusx66p4y",
-        "skill/valory/termination_abci/0.1.0": "bafybeihmqrvnzhptap62nchvocmu7tzl5d2o6k4eitkaiuvo7qwvsaedua"
+        "skill/valory/termination_abci/0.1.0": "bafybeifdtxgldw33kwvsavcituzewwbr6iqfcsgk5qouqfhpwdrivyyyom"
```
Comment (Adamantios) [inline packages/packages.json:43]: ??

Comment (jmoreira-valory) [reply packages/packages.json]: fixed

```
autonomy packages sync --source valory-xyz/open-aea:v1.46.0 --update-packages
autonomy packages sync --source valory-xyz/open-autonomy:0.14.3.post1 --update-packages
```

**Comment by jmoreira-valory** [comment]
Gitleaks is complaining about file `packages/valory/services/trader_omen_gnosis/service.yaml` in another branch.

**Review by Adamantios — APPROVED** [review]

**Pull request merged**